include/picongpu/plugins/transitionRadiation/Particle.hpp

```diff
@@ -0,0 +1,184 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+namespace picongpu
+{
+namespace plugins
+{
+namespace transitionRadiation
+{
+    /** Particle class for transition radiation calculation.
+     *
+     * @param locationSet global position of the macro-particle
+     * @param momentumSet momentum of macro-particle
+     * @param charge
+     */
+    class Particle
+    {
+    private:
+        float3_X const & momentum;
+        float_X const mass;
+        float3_X location;
+        float_X gamma;
+        float3_X beta;
+        float_X betaAbs;
+
+    public:
+        HDINLINE
+        Particle(
+            float3_X const & locationSet,
+            float3_X const & momentumSet,
+            float_X const massSet
+        ) :
+            location( locationSet ),
+            momentum( momentumSet ),
+            mass( massSet )
+        {
+            gamma = calcGamma( );
+            beta = calcBeta( );
+            betaAbs = math::sqrt( ( beta * beta ).sumOfComponents( ) );
+        }
+
+        //! @return momentum
+        HDINLINE
+        float3_X
+        getMomentum( ) const
+        {
+            return momentum;
+        }
+
+        //! @return normalized momentum
+        HDINLINE
+        float_X
+        getU( ) const
+        {
+            return gamma * betaAbs;
+        }
+
+        //! @return velocity v = beta * c
+        HDINLINE
+        float_X
+        getVel( ) const
+        {
+            return betaAbs * picongpu::SPEED_OF_LIGHT;
+        }
+
+        //! propagates the current window to the foil position
+        HDINLINE
+        void
+        propagate( const float_X & propagationDistance )
+        {
+            location += propagationDistance * beta;
+        }
+
+        //! @return polar angle phi of momentum
+        HDINLINE
+        float_X
+        getMomPhi( ) const
+        {
+            // add pi to atan2 function, because phi is in range from 0 to 2 pi
+            return picongpu::math::atan2(
+                momentum.x( ),
+                momentum.z( )
+            ) + picongpu::PI;
+        }
+
+        //! @return azimuth angle psi of momentum
+        HDINLINE
+        float_X
+        getMomTheta( ) const
+        {
+            //because of floating point precision x^2+y^2+z^2<y^2 for x,z<<y
+            float_X const momAbs = getMomAbs( );
+            if( momAbs <= momentum.y( ) )
```

Comment:
This is fine as long our main propagation direction is in positive y-direction - but will fail e.g. for the bunch example where the default propagation direction is in negative y-direction. 
@finnolec What happens if `x^2+y^2+z^2<y^2` is true  but we go into the `else` branch?